### PR TITLE
Add keyword search

### PR DESCRIPTION
**Pull request by brugeman**

### CODE REVIEW

File: src/js/Nostr.ts [hunk 1]
```diff
@@ -87,10 +88,18 @@ const DEFAULT_RELAYS = [
   'wss://brb.io',
 ];
 
+const SEARCH_RELAYS = [
```
Comment (pydlv) [inline src/js/Nostr.ts:91]: why r u hard code this? doesn't this client have settings for it

Comment (mmalmi) [reply src/js/Nostr.ts:91]: Afaik only @brugeman's relay supports keyword search atm

Comment (brugeman) [reply src/js/Nostr.ts:91]: Discussed this w/ @mmalmi 

Only my relay supports keyword search atm. The NIP is proposed, but it won't be considered until some clients implement this. We could tell users that they need to add my relay to their list for search to work, but that means a) it won't really work for anyone, and b) they're forced to save my relay in their contact list and promote it essentially. 

So an approach to hard-code the relay for a while just for search seems better, user's relay lists won't be affected, all users will be able to use the search right away, if some more relays start supporting it we can extend the search relay list, and if NIP is accepted and all relays start supporting it then we drop the hard-coded list entirely.

**Comment by mmalmi** [comment]
Wow, looks like it's working!

**Comment by mmalmi** [comment]
Wonderful, thank you so much @brugeman!

Would you be interested in making it also go through every event in eventsById and show matches from there? (Or at least test if that approach is feasible.) That would work even when you're offline / not reaching your search relay.

**Comment by brugeman** [comment]
> Wonderful, thank you so much @brugeman!
> 
> Would you be interested in making it also go through every event in eventsById and show matches from there? (Or at least test if that approach is feasible.) That would work even when you're offline / not reaching your search relay.

Thanks!

I will look into that.

File: src/js/Nostr.ts (hunk 1 above)
Comment (pydlv) [reply src/js/Nostr.ts:91]: In the interest of privacy and decentralization in that case, this feature should probably live behind a feature flag in settings until NIP gets approved

**Pull request merged**